### PR TITLE
perf: various changes

### PR DESCRIPTION
**Pull request by mattklein123**
1) Don't use fmt::format() in HTTP/1.1 codec hot path
2) Allow x-request-id generation to be disabled
3) Don't do tracing mutation if tracing is not enabled
3) Allow router dynamic stats to be disabled
4) Don't generate useless random numbers for null runtime

### CODE REVIEW

**Comment by mattklein123** [comment]
@lyft/network-team

File: source/common/http/http1/codec_impl.cc [hunk 1]
```diff
@@ -141,7 +147,10 @@ void RequestStreamEncoderImpl::encodeHeaders(const HeaderMap& headers, bool end_
     head_request_ = true;
   }
 
-  output_buffer_.add(fmt::format("{} {} HTTP/1.1\r\n", method, path));
+  output_buffer_.add(method);
+  output_buffer_.add(" ");
+  output_buffer_.add(path);
+  output_buffer_.add(" HTTP/1.1\r\n");
```
Comment (RomanDzhabarov) [inline source/common/http/http1/codec_impl.cc]: use CRLF as well?

Comment (mattklein123) [inline source/common/http/http1/codec_impl.cc]: another function call for no particular reason

Comment (RomanDzhabarov) [inline source/common/http/http1/codec_impl.cc]: ok, sound good

**Pull request merged**